packages/core/src/build/handlers.ts

```diff
@@ -1,15 +1,19 @@
 import { type Message, build, formatMessagesSync } from "esbuild";
 import glob from "glob";
 import { existsSync, rmSync } from "node:fs";
+import { createRequire } from "node:module";
 import path from "node:path";
 import { replaceTscAliasPaths } from "tsc-alias";
 import type { Hex } from "viem";
 
-import type { LogEventMetadata, LogFilter } from "@/config/logFilters";
-import type { Options } from "@/config/options";
-import type { Block } from "@/types/block";
-import type { Log } from "@/types/log";
-import type { Transaction } from "@/types/transaction";
+import type { LogEventMetadata, LogFilter } from "@/config/logFilters.js";
+import type { Options } from "@/config/options.js";
+import type { Block } from "@/types/block.js";
+import type { Log } from "@/types/log.js";
+import type { Transaction } from "@/types/transaction.js";
+import { isRejected } from "@/utils/promise.js";
+
+const require = createRequire(import.meta.url);
```

Comment:
Did you test that hot reloading still works using createRequire with the require cache invalidation trick?

Reply:
yup first thing. Sadly doesn't work

Reply:
m8 be relevant https://github.com/evanw/esbuild/issues/1828